UPGRADE-4.3.md

```diff
@@ -177,3 +177,46 @@ Workflow
        }
    }
    ```
+
+ * `MultipleStateMarkingStore` is deprecated. Use `MethodMarkingStore` instead.
+
+   Before:
+   ```yaml
+   framework:
+       workflows:
+           article:
+               marking_store:
+                   type: multiple
+   ```
+
+   After:
+   ```yaml
+   framework:
+       workflows:
+           article:
+               marking_store:
+                   type: method
+
+   ```
+
+ * `SingleStateMarkingStore` is deprecated. Use `MethodMarkingStore` instead.
+
+   Before:
+   ```yaml
+   framework:
+       workflows:
+           article:
+               marking_store:
+                   type: single
+   ```
+
+   After:
+   ```yaml
+   framework:
+       workflows:
+           article:
+               marking_store:
+                   type: method
+                   arguments:
+                       - true
```

Comment:
I am coming to this late because it was merged yesterday, but in my opinion the DX of this change is not good.

Reading the YAML config file, the meaning of this first argument is not clear (what does `true` mean?).

For specifying a custom property name (as I do) with a multiple state workflow, I now need to remember to do:
   ```yaml
   framework:
       workflows:
           article:
               marking_store:
                   type: method
                   arguments:
                       - false
                       - myColumnName
```

This being YAML, can we use named arguments here? The code would then become:
   ```yaml
   framework:
       workflows:
           article:
               marking_store:
                   type: method
                   arguments:
                       singleState: false
                       property: myColumnName
```

Reply:
I fully agree with that I was about to open an issue for it. We should only rely on the `type` (workflow or state_machine) and don't care about that argument. so we could simplify by:
```yaml
framework:
    workflows:
        article:
            type: state_machine
            marking_store:
                type: method
                property: myColumnName
```

Reply:
Please don't discuss on closed issues/PR

Reply:
Yes I've just opened #30656.